src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -540,25 +540,33 @@ namespace xclhwemhal2 {
       {
         //CR 966701: alignment to 4k (instead of mDeviceInfo.mDataAlignment)
         mDDRMemoryManager.push_back(new xclemulation::MemoryManager(it.size, it.base_addr, getpagesize(), it.tag));
+        std::string tag = it.tag;
+        int32_t index = it.index;
+
+        std::size_t found = tag.find("HOST");
+        if (found != std::string::npos) {
+          host_sptag_idx = index;
+        }
       }
-      for (auto it:mDDRMemoryManager)
+
+      for (auto it : mDDRMemoryManager)
       {
-              std::string tag = it->tag();
+        std::string tag = it->tag();
 
-              if(tag.empty() || tag.find("MBG") == std::string::npos)
-        	      continue;
+        if (tag.empty() || tag.find("MBG") == std::string::npos) {
```

Comment:
How about just:
```
if (it->tag().find("MBG") == std::string::npos)
 ...;
```
find on empty string returns std::string::npos, no need to check empty first.

Reply:
Corrected the condition as suggestion